<?xml version="1.0"?>
<launch>

  <!-- Applies kalman filter on IMU data from k64f_imu node -->
<include file="$(find gpar_k64f)/launch/k64f_imu.launch" >
  <arg name="imu_topic" value="/imu/data_raw" />
  <arg name="mag_topic" default="/imu/mag" />
</include>


<node name="magdwick" pkg="imu_filter_madgwick" type="imu_filter_node" >
<!-- <param name="use_mag" value="false" /> -->
<param name="fixed_frame" value="map" />
<param name="remove_gravity_vector" value="true" />
</node>

 <node pkg="gpar_kalman" type="kalman_k64f" name="kalman_k64f" output="screen" >
 <param name="parent_frame" value="map_tf" />
 <param name="child_frame" value="kalman" />
 </node>

 <node pkg="tf2_ros" type="static_transform_publisher" name="kalman_link" args="1 0 0 0 0 0 map map_tf" />

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find gpar_kalman)/rviz/compare.rviz" />


</launch>
